src/CanvasRenderingContext2d.cc

```diff
@@ -312,8 +312,13 @@ void
 Context2d::fill(bool preserve) {
   if (state->fillPattern) {
     cairo_set_source(_context, state->fillPattern);
-    cairo_pattern_set_extend(cairo_get_source(_context), CAIRO_EXTEND_REPEAT);
-    // TODO repeat/repeat-x/repeat-y
+    repeat_type_t *repeat = static_cast<repeat_type_t *>(cairo_pattern_get_user_data(state->fillPattern, pattern_repeat_key));
```

Comment:
any lint rule here for this long line? any better way to write this thing here?

Reply:
You could do the cast on a separate line perhaps. Also I'd store the pointer's value ASAP.

```cpp
void *ud = cairo_pattern_get_user_data(state->fillPattern, pattern_repeat_key);
repeat_type_t repeat = *reinterpret_cast<repeat_type_t*>(ud);
```

Since you do this twice, you might want to put this in a static method on `Pattern` (like `get_repeat_type_for_cairo_pattern`) and make `pattern_repeat_key` `static const` (better encapsulation).